mmf/datasets/builders/gqa/masked_dataset.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import random
+
+from mmf.common.sample import Sample
+from mmf.datasets.mmf_dataset import MMFDataset
+
+
+class MaskedGQADataset(MMFDataset):
+    def __init__(self, config, dataset_type, imdb_file_index, *args, **kwargs):
+        super().__init__(
+            "masked_gqa", config, dataset_type, imdb_file_index, *args, **kwargs
+        )
+        self._add_answer = config.get("add_answer", True)
+
+    def __getitem__(self, idx):
+        sample_info = self.annotation_db[idx]
+        current_sample = Sample()
+
+        if self._use_features is True:
+            features = self.features_db[idx]
+            current_sample.update(features)
+            image_labels = []
+
+            for i in range(features["image_feature_0"].shape[0]):
```

Comment:
Shouldn't this only be done if the option 'use_image_feature_masks' is enabled? Otherwise, this is unnecessary calculation all of the time. I also think for long term maintenance purpose, we should convert this to a processor.